sdk/communication/azure-communication-chat/azure/communication/chat/_shared/models.py

```diff
@@ -198,11 +255,17 @@ def identifier_from_raw_id(raw_id: str) -> CommunicationIdentifier:
     """
     if raw_id.startswith('4:'):
         return PhoneNumberIdentifier(
-            value = raw_id[len('4:'):]
+            value=raw_id[len('4:'):]
         )
 
     segments = raw_id.split(':', maxsplit=2)
     if len(segments) < 3:
+        if len(segments) == 2 and segments[0] == '28':
+            return MicrosoftBotIdentifier(
+                bot_id=segments[1],
+                is_global=True,
+                cloud=CommunicationCloudEnvironment.PUBLIC
+            )
```

Comment:
272 line -> reuse segments[2] instead of substringing from the raw id

Reply:
The result of this operations could not be consistent if there is another : in the identifier. The current approach is consistent across all platforms and will use the remaining part of the string (no matter if it has : in it or not). The approach you're proposing will only return the third element of the initial list (no matter if there are more elements in it). this might fail in the future with other identifiers.

Reply:
the same response that I left for .net :)

Reply:
Updated